src/browserlib/extract-events.mjs

```diff
@@ -0,0 +1,278 @@
+
+const singlePage = !document.querySelector('[data-reffy-page]');
+
+const href = el => getAbsoluteUrl(el || document.body.querySelector("*[id]"), {singlePage});
+
+export default function (spec) {
+  // Used to find eventhandler attributes
+  const idl = extractWebIdl();
+  const idlTree = parse(idl);
+  const idlInterfaces = idlTree.filter(item => item.type === "interface" || item.type === "interface mixin");
+
+  // associate event names from event handlers to interfaces with such an handler
+  const handledEventNames = idlInterfaces.map(iface => iface.members.filter(m => m.idlType?.idlType === "EventHandler" && m.type === "attribute" && m.name?.startsWith("on")).map(m => [m.name.slice(2), iface.name])).flat().reduce((acc, b) => {
+    if (!acc[b[0]]) acc[b[0]] = [];
+    acc[b[0]].push(b[1]);
+    return acc;
+  }, {});
+
+
+  function fromEventElementToTargetInterfaces(eventEl) {
+    if (!eventEl) return;
+
+    if (eventEl.dataset?.dfnFor || eventEl.dataset?.linkFor) {
+      return (eventEl.dataset.dfnFor || eventEl.dataset.linkFor).split(",").map(t => t.trim());
+    } else if (eventEl.getAttribute("href")?.startsWith("#")) {
+      const dfn = document.getElementById(eventEl.getAttribute("href").slice(1));
+      if (dfn && dfn.dataset?.dfnFor) {
+	return dfn.dataset.dfnFor.split(",").map(t => t.trim());
+      }
+    } else if (handledEventNames[eventEl.textContent]?.length) {
+      // Search for on<event> EventHandler in IDL
+    const matchingInterfaces = handledEventNames[eventEl.textContent];
+      if (matchingInterfaces.length === 1) {
+	// only one such handler, we assume it's a match
+	return matchingInterfaces;
+      } else {
+	console.error("[reffy] Multiple event handler named " + eventEl.textContent + ", cannot associate reliably to an interface in " + spec.title);
+      }
+    }
+  }
+
+
+  let events = [];
+  // Look for event summary tables
+  // ignore DOM spec which uses a matching table format
+  // to map to legacy event types
+  let hasStructuredData = false;
+  if (spec.shortname !== "dom") {
+    document.querySelectorAll("table").forEach(table => {
+      const firstHeading = table.querySelector("thead tr th")?.textContent?.trim();
+      if (firstHeading && firstHeading.match(/^Event/) && firstHeading !== "Event handler") {
+	hasStructuredData = true;
+	// Useful e.g. for pointerevents
+	const bubblingInfoColumn = [...table.querySelectorAll("thead th")].findIndex(n => n.textContent.trim().match(/^bubbl/i));
+	const interfaceColumn = [...table.querySelectorAll("thead th")].findIndex(n => n.textContent.trim().match(/^interface/i));
+	table.querySelectorAll("tbody tr").forEach(tr => {
+	  const event = {};
+	  const eventEl = tr.querySelector("*:first-child");
+	  const annotations = eventEl.querySelectorAll("aside, .mdn-anno");
+	  annotations.forEach(n => n.remove());
```

Comment:
In [`extract-cssdfn.mjs`](https://github.com/w3c/reffy/blob/main/src/browserlib/extract-cssdfn.mjs#L56), extract is cloned before annotations get removed not to modify the DOM that further browser modules may need to parse. Who knows, we may want to add a module that parses the aside and MDN annotations at some point.

Reply:
thanks, added a code suggestion to that effect